### ISSUE TITLE
Facebook Search Deprecation

### ISSUE THREAD

**Issue by coderhs** [arsduo/koala#400]
I ran the following code today as part of testing scumblr and got the following error (https://github.com/Netflix/Scumblr/issues/11)

```ruby
        app_id = Rails.configuration.try(:facebook_app_id)
        app_secret = Rails.configuration.try(:facebook_app_secret)
        oauth = Koala::Facebook::OAuth.new(app_id, app_secret)
        @access_token = oauth.get_app_access_token
        @graph = Koala::Facebook::API.new(@access_token)
        @graph.search(ruby) 
```

The last line creates this error. "ERROR: type: OAuthException, code: 11, message: (11) Post search has been deprecated [HTTP 400]"

I am using Koala 1.10.0

**Comment by Clee681** [comment]
@arsduo I think you may have prematurely removed the search functionality. From [FB's changelog](https://developers.facebook.com/docs/apps/changelog), Seems like "Public Post" search was deprecated, but there are still cases where the 'search' endpoint is used (e.g. [ad targeting search](https://developers.facebook.com/docs/marketing-api/targeting-search/v2.3)). Any chance you can revert https://github.com/arsduo/koala/commit/01db4a9089a59b00c224e083f057b825e75584fd?

**Comment by billvieux** [comment]
@arsduo @Clee681 is correct. I still search for pages with 
`api.search(query, { :type => 'page' })`
In [Graph Explorer](https://developers.facebook.com/tools/explorer/) you can confirm
`/search?q=test&type=page`
is still a valid endpoint on the 2.x API.

**Comment by jdgblinq** [comment]
+1 revert

**Comment by guyisra** [comment]
+1 search is still available in facebook v2.3 so that like up to year 2017
https://developers.facebook.com/docs/graph-api/using-graph-api/v2.3#search

**Comment by guyisra** [comment]
@arsduo is it possible to bring this back in v 2.2?

**Comment by arsduo** [comment]
Yes -- my apologies for making the mistake and for the fix taking so long.
I'll take care of it this week, unless someone can get to it first.

2015-08-02 13:35 GMT-05:00 guyisra <notifications@github.com>:

**Comment by arsduo** [comment]
Just merged in a pull request that restores search! (Thanks, @vhoof!) This has been released as koala 2.2.0rc3 -- can you test it out and see if you encounter any issues?

Thanks!
